jbmc/unit/java_bytecode/java_bytecode_parser/parse_java_annotations.cpp

```diff
@@ -0,0 +1,90 @@
+/*******************************************************************\
+
+ Module: Unit tests for converting annotations
+
+ Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include <java-testing-utils/load_java_class.h>
+#include <testing-utils/catch.hpp>
+#include <java_bytecode/java_types.h>
+#include <java_bytecode/java_bytecode_parse_tree.h>
+#include <java_bytecode/java_bytecode_convert_class.h>
+
+// See
+// https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.7.16.1
+SCENARIO(
+  "java_bytecode_parse_annotations",
+  "[core][java_bytecode][java_bytecode_parser]")
+{
+  GIVEN("Some class files in the class path")
+  {
+    WHEN("Parsing an annotation with Class value specified to non-primitive")
+    {
+      const symbol_tablet &new_symbol_table = load_java_class(
+        "ClassWithClassTypeAnnotation",
+        "./java_bytecode/java_bytecode_parser");
+
+      THEN("The annotation should store the correct type")
+      {
+        const symbolt &class_symbol =
+          *new_symbol_table.lookup("java::ClassWithClassTypeAnnotation");
+        const std::vector<java_annotationt> &java_annotations =
+          to_annotated_type(class_symbol.type).get_annotations();
+        java_bytecode_parse_treet::annotationst annotations;
+        convert_java_annotations(java_annotations, annotations);
+        REQUIRE(annotations.size() == 1);
+        const auto &annotation = annotations.front();
+        const auto &element_value_pair = annotation.element_value_pairs.front();
+        const auto &id = to_symbol_expr(element_value_pair.value)
+          .get_identifier();
+        REQUIRE(id2string(id) == "Ljava/lang/String;");
```

Comment:
I would prefer having the CPROVER format for java object types here

Reply:
Where can I find that? And similar to my above question: I'm not sure why using hard-coded values in unit tests is a problem?

Reply:
effectively it would be `java::java.lang.String` in this case, not the raw JVM string. The `java_type_from_string` method should return a symbol type with the correct identifier in it.

Reply:
The problem with `java_type_from_string` is that it errors if the class is not actually loaded. I'm looking for an alternative that just converts the string without checking if the type is known...

Reply:
Changed it to use `java_type_from_string` and then compare the string in that type with `java::java.langString`. Does that make sense?
(The mistake I made before was to run gdb on the unit test, which randomly fails for me on various tests for some reasons even though the tests pass when I run them normally...)